### PR TITLE
Upgrade to Gulp 4

### PR DESCRIPTION
**Pull request by dweinber**
Newer node versions started to fail running gulp3. This PR upgrades the build chain to gulp4 based on https://www.liquidlight.co.uk/blog/how-do-i-update-to-gulp-4/

Issue: https://github.com/bitmovin/bitmovin-player-ui/issues/208

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -199,28 +193,22 @@ gulp.task('sass', function() {
...
-    callback);
-});
+gulp.task('build', gulp.series('clean', gulp.parallel('html', 'browserify', 'sass')));
 
-gulp.task('build-prod', function(callback) {
-  production = true;
-  runSequence('lint', 'build', callback);
-});
+gulp.task('build-prod', gulp.series(function(callback) {
+  production = true
```
Comment (protyposis) [inline gulpfile.js]: missing semicolon

File: gulpfile.js [hunk 2]
```diff
@@ -199,28 +193,22 @@ gulp.task('sass', function() {
...
+  callback();
```
Comment (protyposis) [inline gulpfile.js]: I think the callback is not necessary (no async code)

Comment (dweinber) [reply gulpfile.js]: I thought the same and I didn't have it in the first place. 

But if I change it to:
```ts
gulp.task('build-prod', gulp.series(function() {
  production = true;
}, 'lint', 'build'));
```
I get an error:
```
$ gulp build-prod
Using gulpfile ~/Projects/bitmovin-player-ui/gulpfile.js
Starting 'build-prod'...
Starting '<anonymous>'...
The following tasks did not complete: build-prod, <anonymous>
Did you forget to signal async completion?
```

Therefore, I added the callback and it works fine then. Happy to change it if we can find out what the problem is, but currently I don't really have an idea.

**Comment by dweinber** [comment]
The `<anonymous>` tasks are all the anonymous functions used in `gulp.serial` or `gulp.parallel` calls.

File: gulpfile.js [hunk 3]
```diff
@@ -116,11 +114,7 @@ gulp.task('lint-sass', function() {
 });
 
 // Runs all linters
-gulp.task('lint', function(callback) {
-  // this fails at first error so we can't run all linters sequentially with runSequence and then print the errors
-  runSequence('lint-ts', 'lint-sass', callback);
-  // TODO check in Gulp 4.0 if all linters can be run sequentially before aborting due to an error
-});
+gulp.task('lint', gulp.series('lint-ts', 'lint-sass'));
```
[the remark this answers is not in the document]
Comment (dweinber) [reply gulpfile.js]: I'll have another look.

Now that I think about this - why do we have to run them sequentially as stated in the comment? Can't we run them in parallel?

File: gulpfile.js (hunk 2 above)
Comment (dweinber) [reply gulpfile.js]: Missed that part, thanks!

File: gulpfile.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (dweinber) [reply gulpfile.js]: This would output both, sass and ts lint errors/warning.

**Pull request merged**